hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueHealth.kt

```diff
@@ -84,10 +95,13 @@ class HmppsQueueHealth(private val hmppsQueue: HmppsQueue) : HealthIndicator {
     }
   }
 
-  private fun getDlqAttributes(): Result<GetQueueAttributesResult> =
-    runCatching {
-      hmppsQueue.sqsDlqClient.getQueueAttributes(GetQueueAttributesRequest(hmppsQueue.dlqUrl).withAttributeNames(All))
-    }
+  private fun getDlqAttributes(): Result<GetQueueAttributesResult> {
+    return if (hmppsQueue.sqsDlqClient == null) failure(RuntimeException("Attempted to access dlqclient that does not exist"))
+    else
+      runCatching {
+        hmppsQueue.sqsDlqClient!!.getQueueAttributes(GetQueueAttributesRequest(hmppsQueue.dlqUrl).withAttributeNames(All))
+      }
+  }
```

Comment:
You could make this a bit prettier:
```
runCatching {
  hmppsQueue.sqsDlqClient
    ?.getQueueAttributes(GetQueueAttributesRequest(hmppsQueue.dlqUrl).withAttributeNames(All))
    ?: throw RuntimeException("Attempted to access dlqclient that does not exist")
}
```
And for the exception I'd probably include the `dlqName` in the message (presumably there is one or we wouldn't have got this far).  And make it custom exception that extends RuntimeException - generally good practice.

Reply:
yes.. not the neatest.  Will update.  Was wondering about the customeexception - so cheers. will add that too